dashboard/api.py

```diff
@@ -21,26 +22,69 @@ def prepend_urls(self):
                 self.wrap_view('get_equipment'), name='api_get_equipment'),
             url(r"^(?P<equipment_resource>%s)/switch%s$" %
                 (self._meta.equipment_resource, trailing_slash()),
-                self.wrap_view('switch_equipment'), name='api_switch_equipment'),
+                self.wrap_view('toggle_equipment'), name='api_toggle_equipment'),
             url(r"^(?P<user_resource>%s)/login%s$" %
                 (self._meta.user_resource, trailing_slash()),
                 self.wrap_view('validate_user'), name='api_validate_user')
         ]
 
+    def validate_key(self, body, key):
+        if not body:
+            result = {'status':False, 'message': f'Expected equipment {key}'}
+        equipment = Equipment.objects.filter(id=body[key])
+        if equipment.count() < 1:
+            result = {'status':False, 'message': f'Equipment {key} does not exist'}
+        return {'status': True, 'query': equipment[0]}
+
     def get_equipment(self, request, *args, **kwargs):
-        result = {'name': 'tubelight', 'rating': 150}
-        return self.create_response(request, result)
+        body = json.loads(request.body)
+        result = self.validate_key(body, 'id')
+        if not result['status']:
+            return self.create_response(request, result)
+        equipment = result['query']
+        response = {
+            'name': equipment.name,
+            'rating': equipment.rating,
+            'priority': equipment.priority
+        }
+        return self.create_response(request, response)
 
-    def validate_user(self, request, *args, **kwargs):
+    def toggle_equipment(self, request, *args, **kwargs):
+        # equip_id, status
         body = json.loads(request.body)
-        # think about good authentication module
-        result = {'status': True, 'body': body}
+        result = self.validate_key(body, 'id')
+        if not result['status']:
+            return self.create_response(request, result)
+        equipment = result['query']
+        equipment_usage = Usage.objects.filter(equipment=equipment)
+        if equipment_usage.count() < 1:
+            result = {'status':False, 'message': f'{equipment.name}\'s usage does not exist'}
+            return self.create_response(request, result)
+        equipment_usage = equipment_usage[0]
+        required_state = body.get('state')
+        if equipment_usage.state and required_state:
```

Comment:
Don't worry about this condition, we can use a toggle switch to evade such problems